api/app/Models/PoolCandidate.php

```diff
@@ -101,25 +101,26 @@ public function scopeClassifications(Builder $query, array $classifications): Bu
 
         // Classifications act as an OR filter. The query should return candidates with any of the classifications.
         // A single whereHas clause for the relationship, containing multiple orWhere clauses accomplishes this.
-
-        // group these in a subquery to properly handle "OR" condition
+        // mirroring functionality of the classification scope on the User model
         $query->where(function ($query) use ($classifications) {
-            $query->whereHas('expectedClassifications', function ($query) use ($classifications) {
-                foreach ($classifications as $index => $classification) {
-                    if ($index === 0) {
-                        // First iteration must use where instead of orWhere
-                        $query->where(function ($query) use ($classification) {
-                            $query->where('group', $classification['group'])->where('level', $classification['level']);
-                        });
-                    } else {
-                        $query->orWhere(function ($query) use ($classification) {
-                            $query->where('group', $classification['group'])->where('level', $classification['level']);
-                        });
+            $query->whereHas('user', function ($query) use ($classifications) {
```

Comment:
FYI, this would break the search page which is currently live in Production, since those PoolCandidates don't have users with complete profiles.
I think its safe to commit this right now, but only because we're planning on turning the APPLICANTSEARCH feature flag on and [migrating candidate data into the user objects](https://github.com/GCTC-NTGC/gc-digital-talent/issues/3949) with the next deployment.